### PR TITLE
Fix query parameters for external links

### PR DESCRIPTION
**Pull request by ghost**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| License | MIT

The external link type has removed all query parameters before editing.

Video before:

https://user-images.githubusercontent.com/47974249/187246980-40eeae23-bade-41d8-93dd-2f3dabbd67ea.mov

Video after:

https://user-images.githubusercontent.com/47974249/187247051-d517d40e-e767-4526-a91c-35bc95a03533.mov

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Link/overlays/ExternalLinkTypeOverlay.js [hunk 1]
```diff
@@ -42,6 +42,12 @@ class ExternalLinkTypeOverlay extends React.Component<LinkTypeOverlayProps> {
             return;
         }
 
+        if (this.protocol !== 'mailto:') {
```
Comment (luca-rath) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/Link/overlays/ExternalLinkTypeOverlay.js]:
```suggestion
        if (!href.startsWith('mailto:')) {
```

**Review by luca-rath — CHANGES_REQUESTED** [review]
Hi, thanks a lot for the pull request. Unfortunately your solution breaks the current behaviour of `mailto` links, because `this.protocol` is not yet set at that point. It will be set with a callback from the `Url` component, as soon as that renders, but the suggestion in my comment fixes that problem.

There are two things that prevent me from merging this right now:
- Please rebase this to the `2.4` branch. We follow SEMVER, so we fix bugs in the lowest supported version (2.4) and introduce new features in the next minor version (2.6)
- It would be great if you could test this fix. In the `ExternalLinkTypeOverlay.test.js` is already a `Render overlay with a URL` test, your solution could be easily tested, if you just change the url in that test from `http://www.sulu.io` to something like `http://www.sulu.io?foo=bar`. But be aware, that you also have to update the snapshot file then (you can run the tests for that file and update the snapshots using `npm run test -- -i ExternalLinkTypeOverlay.test.js -u`)

**Comment by niklasnatter** [comment]
Hey, thanks for the pull request! 
I have fixed the handling of the `mailto:` specific parameters and added two respective test cases 🙂

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Link/tests/overlays/ExternalLinkTypeOverlay.test.js [hunk 2]
```diff
@@ -84,6 +84,27 @@ test('Pass correct props to Dialog', () => {
     expect(externalLinkOverlay.find('Dialog').prop('open')).toEqual(false);
 });
 
+test('Display given URL with query parameters in href input', () => {
+    const targetChangeSpy = jest.fn();
+    const urlChangeSpy = jest.fn();
+
+    const externalLinkOverlay = shallow(
+        <ExternalLinkTypeOverlay
+            href="http://www.sulu.io?param=value-1"
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/Link/tests/overlays/ExternalLinkTypeOverlay.test.js]:
```suggestion
            href="http://www.sulu.io/contact-us?param=value-1"
```

use a valid url. Domain without `/` is not a valid url it always requires a path part and for homepage it is `/`.

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Link/tests/overlays/ExternalLinkTypeOverlay.test.js [hunk 3]
```diff
@@ -84,6 +84,27 @@ test('Pass correct props to Dialog', () => {
...
+            onHrefChange={urlChangeSpy}
+            onTargetChange={targetChangeSpy}
+            onTitleChange={jest.fn()}
+            open={true}
+            target="_blank"
+            title={undefined}
+        />
+    );
+
+    expect(externalLinkOverlay.find('Url').prop('value')).toEqual('http://www.sulu.io?param=value-1');
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/Link/tests/overlays/ExternalLinkTypeOverlay.test.js]:
```suggestion
    expect(externalLinkOverlay.find('Url').prop('value')).toEqual('http://www.sulu.io/contact-us?param=value-1');
```

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Link/tests/overlays/ExternalLinkTypeOverlay.test.js (hunk 3 above)
Comment (niklasnatter) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/Link/tests/overlays/ExternalLinkTypeOverlay.test.js]: pushed the change together with linting fixes 🙂

**Review by alexander-schranz — APPROVED** [review]

**Comment by alexander-schranz** [comment]
@flogado @niklasnatter Thank you!

**Pull request merged**